Tests/Core/AWSClientRuntimeTests/Sigv4/SigV4SigningTests.swift

```diff
@@ -17,6 +18,278 @@ class Sigv4SigningTests: XCTestCase {
+        signingProperties.set(key: AttributeKeys.unsignedBody, value: false)
+        signingProperties.set(key: AttributeKeys.signingRegion, value: "us-east-1")
+        signingProperties.set(key: AttributeKeys.signingAlgorithm, value: .sigv4)
+        signingProperties.set(key: AttributeKey<Date>(name: "SigV4AuthSchemeTests"), value: date)
+
+        do {
+            _ = try await AWSSigV4Signer().signRequest(
+                requestBuilder: requestBuilder,
+                identity: credentials,
+                signingProperties: signingProperties
+            )
+            XCTFail("The code failed to throw at expected point.")
+        } catch ClientError.authError(let message) {
+            let expectedMessage = "Signing properties passed to the AWSSigV4Signer must contain signing name."
+            XCTAssertEqual(expectedMessage, message)
+        } catch {
+            XCTFail("The code did not throw expected type of error.")
+        }
+    }
+
+    func testSignRequestMissingSigningRegion() async throws {
+        let dateString = "2024-01-16T12:36:00Z"
+        guard let date = TimestampFormatter(format: .dateTime).date(from: dateString) else {
+            XCTFail("Unable to parse date")
+            return
+        }
+
+        let requestBuilder = SdkHttpRequestBuilder()
+            .withHost("example.amazonaws.com")
+            .withPath("/")
+            .withMethod(.get)
+            .withPort(443)
+            .withProtocol(.http)
+            .withHeader(name: "host", value: "example.amazonaws.com")
+
+        let credentials = AWSCredentials(accessKey: "test-access-key", secret: "test-secret-key")
+
+        var signingProperties = Attributes()
+        signingProperties.set(key: AttributeKeys.bidirectionalStreaming, value: false)
+        signingProperties.set(key: AttributeKeys.unsignedBody, value: false)
+        signingProperties.set(key: AttributeKeys.signingName, value: "test")
+        signingProperties.set(key: AttributeKeys.signingAlgorithm, value: .sigv4)
+        signingProperties.set(key: AttributeKey<Date>(name: "SigV4AuthSchemeTests"), value: date)
+
+        do {
+            _ = try await AWSSigV4Signer().signRequest(
+                requestBuilder: requestBuilder,
+                identity: credentials,
+                signingProperties: signingProperties
+            )
+            XCTFail("The code failed to throw at expected point.")
+        } catch ClientError.authError(let message) {
+            let expectedMessage = "Signing properties passed to the AWSSigV4Signer must contain signing region."
+            XCTAssertEqual(expectedMessage, message)
+        } catch {
+            XCTFail("The code did not throw expected type of error.")
+        }
+    }
+
+    func testSignRequestMissingSigningAlgorithm() async throws {
+        let dateString = "2024-01-16T12:36:00Z"
+        guard let date = TimestampFormatter(format: .dateTime).date(from: dateString) else {
```

Comment:
As an alternative this guard statement can be a one-line `XCTUnwrap()`:
```swift
let date = try XCTUnwrap(TimestampFormatter(format: .dateTime).date(from: dateString))
```
If `date` is nil, `XCTUnwrap()` will throw, causing the test to fail.

Reply:
Changed as suggested for concise code.